pyomo/contrib/solver/common/base.py

```diff
@@ -638,11 +674,16 @@ def license_is_valid(self) -> bool:
         return bool(self.available())
 
     def config_block(self, init=False):
-        from pyomo.scripting.solve_config import default_config_block
-
+        """
+        Preserves config backwards compatibility; allows new solver interfaces
+        to be used in the pyomo solve call
+        """
         return default_config_block(self, init)[0]
 
     def set_options(self, options):
+        """
+        Method to manually set options; can be called outside of the solve method
+        """
         opts = {k: v for k, v in options.value().items() if v is not None}
```

Comment:
This implementation implicitly assumes `options` is a `ConfigDict`.  We should update the documentation to include argument type specification, and then update the implementation to accept `dict` or `ConfigDict` (by dropping the call to `value()`:
```python
        opts = {k: v for k, v in options.items() if v is not None}
```